### PR TITLE
xed/mbuild: cleanup

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: pkgs/xed/default.nix [hunk 1]
```diff
@@ -10,24 +10,17 @@ stdenv.mkDerivation rec {
...
-  nativeBuildInputs = [ python ];
-
-  buildPhase = ''
-    python ./mfile.py -j$NIX_BUILD_CORES
-  '';
+  nativeBuildInputs = [ mbuild ];
 
   installPhase = ''
-    python ./mfile.py -j$NIX_BUILD_CORES --prefix=$out install
+    python ./mfile.py install --shared --install-dir=$out
```
Comment (Mic92) [inline pkgs/xed/default.nix:16]: just let me no if `--shared` works for you

**Comment by Mic92** [comment]
I have this package too in my repository. But I will probably not update it, since I no longer need it.
Here is the stuff I had in addition to your expression.

File: pkgs/xed/default.nix [hunk 2]
```diff
@@ -10,24 +10,17 @@ stdenv.mkDerivation rec {
...
+    ln -s $out/include/xed/* $out/include
```
Comment (Mic92) [inline pkgs/xed/default.nix:17]: Intel guys are using it slightly different: https://github.com/01org/processor-trace/blob/master/ptxed/src/ptxed.c#L47

Comment (dtzWill) [reply pkgs/xed/default.nix:17]: Hmm, interesting.

Did some digging and found this: https://github.com/intelxed/xed/commit/fcd2d41b2bd8802eed4e620a6836795fbd7ad919#diff-4c830633847d5e1220d61fab27a0e28a

So I think it should be left as-is, and do as they say re:adjusting include path or fixing code.

**Comment by dtzWill** [comment]
Awesome! Going through this now!

**Comment by dtzWill** [comment]
Thanks! Gonna add some options I need for another package, but will build on this.

File: pkgs/xed/default.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/xed/default.nix:17]: I guess I will send a patch for processor-trace if I use the ptxed in future.

**Pull request merged**